File: src-tauri/src/models/categoria.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Categoria {
  pub cat_uuid: String,
  pub usu_uuid: String,
  pub sit_uuid: String,
  pub cat_descricao: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategoriaCriar {
  pub cat_descricao: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategoriaAtualizar {
  pub cat_uuid: String,
  pub cat_descricao: String,
}

File: lib/models/categoria.ts
export interface Categoria {
  cat_uuid: string;
  usu_uuid: string;
  sit_uuid: string;
  cat_descricao: string;
}

export interface CategoriaCriar {
  cat_descricao: string;
}

export interface CategoriaAtualizar {
  cat_uuid: string;
  cat_descricao: string;
}
